pkg/tools/gen/templates/kcl/config.gotmpl

```diff
@@ -0,0 +1,29 @@
+{{- if .Var }}{{ formatName .Var }}{{- if .IsUnion }}{{ " : " }}{{- else }}{{ " = " }}{{- end }}{{- end }}{{- if .Name }}{{ formatName .Name }}{{- end }}{{- if .Data }}{{- "{\n" }}
+{{- range .Data -}}
+{{- "    " }}{{ formatName .Key }}{{ " = " }}
+    {{- if isKclData .Value }}
```

Comment:
duplicate with `data.gotmpl`. we can replace it with`{{- indentLines (include "data"  .) "    " }}`

Reply:
> duplicate with `data.gotmpl`. we can replace it with`{{- indentLines (include "data" .) " " }}`

This is a good suggestion, but the four leading spaces of `}` and `]` have been added to hardcode in `data.tmpl`. Thus, the `config.tmpl` and `data.tmpl` should me merged in the future. 😄

Reply:
Yes, we can merge them in the future. However I am a little confuse about what you meant by saying "the four leading spaces of } and ] have been added to hardcode in data.tmpl."
I modify the `config.gotmpl` to the code below and all tests pass:
```
{{- if .Var }}{{ formatName .Var }}{{- if .IsUnion }}{{ " : " }}{{- else }}{{ " = " }}{{- end }}{{- end }}{{- if .Name }}{{ formatName .Name }}{{- end }}{{- if .Data }}{{- "{\n" }}
{{- range .Data -}}
{{- indentLines (include "data"  .) "    " }}
{{- end -}}
{{- "}" }}{{- else }}{}{{- end }}
```
Did I miss anything? I would appreciate it if you could provide a little more detail about it. Thanks❤

Reply:
You are right, I was wrong. I have modified the corresponding code.